### PR TITLE
[HOTFIX] Logging for evaluation

### LINKED ISSUE #4681: self.log on validation_step is broken on pre 1.1 [nightly]
https://colab.research.google.com/drive/1tSphAIaCdy3tC9Lzhe1GEK_fH_0a6oYj?usp=sharing

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR resolve a missing self.log_metrics called at the end of an epoch !

Closes #4681 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In in short, see following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py [hunk 1]
```diff
@@ -258,6 +260,11 @@ def prepare_eval_loop_results(self):
             self.add_to_eval_loop_results(dl_idx, has_been_initialized)
 
     def get_evaluate_epoch_results(self, test_mode):
+        if not self.trainer.running_sanity_check:
+            # log all the metrics as a s single dict
```
Comment (SeanNaren) [inline pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py:264]:
```suggestion
            # log all the metrics as a single dict
```

File: tests/trainer/logging_tests/test_eval_loop_logging_1_0.py [hunk 2]
```diff
@@ -780,3 +783,95 @@ def get_expected_output(func_attr, original_values):
...
+        limit_test_batches=0,
+        max_epochs=2,
+        progress_bar_refresh_rate=1,
+    )
+
+    # Train the model ⚡
+    trainer.fit(model)
+
+    # hp_metric + 2 steps + epoch + 2 steps + epoch
+    expected = 1 + 2 + 1 + 2 + 1
```
Comment (SeanNaren) [inline tests/trainer/logging_tests/test_eval_loop_logging_1_0.py:839]: Let's be a bit more clearer here, maybe `expected_num_calls`

File: tests/trainer/logging_tests/test_eval_loop_logging_1_0.py [hunk 3]
```diff
@@ -780,3 +783,95 @@ def get_expected_output(func_attr, original_values):
...
+
+    assert len(mock_log_metrics.mock_calls) == expected
+
+    assert mock_log_metrics.mock_calls[0] == call({'hp_metric': -1}, 0)
+
+    def get_keys_at_idx(idx):
+        keys = mock_log_metrics.mock_calls[idx].kwargs["metrics"].keys()
+        return sorted([k for k in keys])
+
+    excepted = ['valid_loss_0_step/epoch_0', 'valid_loss_2/epoch_0', 'global_step']
```
Comment (SeanNaren) [inline tests/trainer/logging_tests/test_eval_loop_logging_1_0.py:849]: I think this should be `expected` not `excepted`. Will need to fix keys below

**Comment by williamFalcon** [comment]
looks good. does this also log to the prog bar?

**Comment by tchaton** [comment]
> looks good. does this also log to the prog bar?

Yes, it should, but I will make sure of this on Monday.

**Pull request merged**